crates/orchestrator/src/telemetry.rs

```diff
@@ -0,0 +1,138 @@
+use once_cell::sync::Lazy;
+use opentelemetry::global;
+use opentelemetry::metrics::Meter;
+use opentelemetry::trace::TracerProvider;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::ExportConfig;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::metrics::reader::DefaultAggregationSelector;
+use opentelemetry_sdk::metrics::reader::DefaultTemporalitySelector;
+use opentelemetry_sdk::metrics::PeriodicReader;
+use opentelemetry_sdk::metrics::SdkMeterProvider;
+use opentelemetry_sdk::trace::BatchConfigBuilder;
+use opentelemetry_sdk::trace::Config;
+use opentelemetry_sdk::trace::Tracer;
+use opentelemetry_sdk::{runtime, Resource};
+use std::sync::Arc;
+use std::time::Duration;
+use utils::env_utils::get_env_var_or_panic;
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    #[derive(Debug)]
+    pub static ref OTEL_SERVICE_NAME: String = get_env_var_or_panic("OTEL_SERVICE_NAME");
+    #[derive(Debug)]
+    pub static ref OTEL_COLLECTOR_ENDPOINT: String = get_env_var_or_panic("OTEL_COLLECTOR_ENDPOINT");
+}
+
+static METER_PROVIDER: Lazy<Arc<SdkMeterProvider>> = Lazy::new(|| {
+    let meter_provider = init_metrics();
+    global::set_meter_provider(meter_provider.clone());
+    Arc::new(meter_provider)
+});
+
+static TRACER: Lazy<Arc<Tracer>> = Lazy::new(|| {
+    let tracer = init_tracer_provider();
+    Arc::new(tracer)
+});
+
+static GLOBAL_METER: Lazy<Meter> = Lazy::new(|| {
+    let common_scope_attributes = vec![KeyValue::new("crate", "orchestrator")];
+
+    global::meter_with_version(
+        "orchestrator.opentelemetry",
+        // TODO: Unsure of these settings, come back
+        Some("0.17"),
+        Some("https://opentelemetry.io/schemas/1.2.0"),
+        Some(common_scope_attributes.clone()),
+    )
+});
+
+pub fn init() {
+    // Force initialization of METER_PROVIDER and GLOBAL_METER
+    Lazy::force(&METER_PROVIDER);
+    Lazy::force(&GLOBAL_METER);
+    Lazy::force(&TRACER);
+}
+
+pub fn global_meter() -> &'static Meter {
+    &GLOBAL_METER
+}
+
+pub fn global_meter_provider() -> &'static SdkMeterProvider {
+    &METER_PROVIDER
+}
+
+pub fn global_tracer() -> &'static Tracer {
+    &TRACER
+}
+
+pub fn init_tracer_provider() -> Tracer {
+    let batch_config = BatchConfigBuilder::default()
+    // Increasing the queue size and batch size, only increase in queue size delays full channel error.
+    .with_max_queue_size(10000)
+    .with_max_export_batch_size(512) // On default
```

Comment:
do we need this now, does default size work now?

Reply:
Yes, this works on default values as well now, removed these setters [here](https://github.com/madara-alliance/madara-orchestrator/pull/131/commits/7d15f65dc99946ca73c0853f48fcf5ca024386ee).